### PR TITLE
Fix/make file-imput more accessible and use semantic html

### PR DESCRIPTION
**Pull request by doriengr**
- [x] Fix bug on delete button
- [x] Add check if file is really `.csv`

Close green-ecolution/green-ecolution-frontend#46

### CODE REVIEW

**Comment by Mohammad96Assaf** [comment]
As I mentioned in #98, the reason I originally used a custom style was to solve the localization issue, as I couldn't control the text in the default browser input. I also have another question: what about the progress bar that displays the upload status? Why did you remove it? :)

**Comment by Mohammad96Assaf** [comment]
> > As I mentioned in #98, the reason I originally used a custom style was to solve the localization issue, as I couldn't control the text in the default browser input. I also have another question: what about the progress bar that displays the upload status? Why did you remove it? :)
> 
> I removed it becauce the selection of a file does never take long and its not needed in my eyes for some couple of seconds. We could implement a process bar while the Import is running.
> 
> And the localization works in my eyes

The progress bar actually demonstrates the import process. It runs until the server returns a response. Here's the code snippet:

```
const pr = await axios.post(to, formData, {
    headers: {
        'Authorization': authHeader,
        'Content-Type': 'multipart/form-data',
    }, 
    onUploadProgress: (progressEvent) => {
        if (progressEvent.total) {
            const percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total);
            setProgress(percentCompleted);
        }
    }
});
```
Sometimes the process might take more than just a few seconds, so in my opinion, the progress bar helps provide feedback to the user. Also, in the backend, the upload process is not yet separated from the import process, which is why the progress bar reflects the entire operation.

File: frontend/package.json [hunk 1]
```diff
@@ -47,8 +47,7 @@
     "tailwind-merge": "^2.3.0",
     "tailwindcss-animate": "^1.0.7",
     "zod": "^3.23.8",
-    "zustand": "^4.5.4",
-    "axios": "^1.7.7"
```
Comment (choffmann) [inline frontend/package.json:51]: :+1:

File: frontend/src/routes/_protected/settings/import.tsx [hunk 2]
```diff
@@ -51,15 +100,35 @@ function ImportFile() {
...
-        <label className="block text-sm font-medium text-gray-700 mb-2">
-          CSV-Datei mit aktuellen Bäumen:
-        </label>
-        <FileUpload to={url} fileType='.csv'/>
+        <Modal
+          title="Soll der Import wirklich neu angestoßen werden?"
+          description="Der Import kann etwas länger dauern, sodass die Website für einen Moment in den Wartungsmodus schaltet und nicht erreichbar ist."
+          confirmText="Import fortfahren"
+          onConfirm={handleConfirm}
+          onCancel={() => console.log('test')}
```
Comment (choffmann) [inline frontend/src/routes/_protected/settings/import.tsx:129]: Close Modal?

```suggestion
          onCancel={() => setIsModalOpen(false)}
```

File: frontend/src/routes/_protected/settings/import.tsx [hunk 3]
```diff
@@ -1,13 +1,62 @@
...
+      const fullErrorMessage = additionalMessage ? `${errorMessage}: ${additionalMessage}` : errorMessage;
+
+      setMessage('Leider ist folgender Fehler aufgetreten: ' + fullErrorMessage);
+    } else if (message) {
+      setMessage('Leider ist folgender Fehler aufgetreten: ' + message);
+    } else {
+      const message = 'Leider ist ein unbekannter Fehler aufgetreten. Bitte wenden Sie sich an eine:n Systemadministrator:in.';
+      setMessage(message);
+    }
+  };
```
Comment (choffmann) [inline frontend/src/routes/_protected/settings/import.tsx:56]: I would prefer to use the query client here with the functions coming from the backend client. That way we don't have to deal with error messages, load state and so on ourselves.

File: frontend/src/components/general/fileUpload/FileUpload.tsx
Comment (choffmann) [inline frontend/src/components/general/fileUpload/FileUpload.tsx:1]: Should we check here that the file is actually a csv file? Yes, you can pass a `fileType` from the browser to the file explorer for an initial filter, which is done here, but you can override this and select a different file type. 

Eingefügtes Bild

**Review by choffmann — APPROVED** [review]
:+1:

**Pull request merged**